engine/runtime-parser/src/main/java/org/enso/compiler/core/TreeToIr.java

```diff
@@ -49,7 +51,14 @@ final class TreeToIr {
   static final String SKIP_MACRO_IDENTIFIER = "SKIP";
   static final String FREEZE_MACRO_IDENTIFIER = "FREEZE";
 
+  private final Map<Location, UUID> idMap;
+
   private TreeToIr() {
+    this.idMap = Collections.emptyMap();
+  }
+
+  public TreeToIr(Map<Location, UUID> idMap) {
```

Comment:
- remove `public` and make `private`
- create a _factory method_
- remove `MODULE` constant and use the _factory method_